### PR TITLE
New design (UI) - appbar color

### PR DESCRIPTION
**Pull request by JorisBodin**
Hello,

I finished changing the color of the appbar.
I also changed the progressbar to match the new design.

newAppBar

Next week I'll continue on the appbar. I will change in the pictures view and search. When I'm done I'll change its style.

See all futur design: https://github.com/nextcloud/android/issues/5207#issuecomment-617270183

### CODE REVIEW

**Comment by JorisBodin** [comment]
@jancborchardt The "Search" placeholder has not changed. It's the native android. It should change :)

File: src/main/java/com/owncloud/android/ui/activity/ToolbarActivity.java [hunk 1]
```diff
@@ -166,25 +160,14 @@ protected final void hideInfoBox() {
...
-        if (mProgressBar != null) {
-            mProgressBar.setIndeterminate(indeterminate);
-        }
-    }
-
     /**
      * Change the visibility for the toolbar's progress bar.
      *
-     * @param visibility visibility of the progress bar
+     * @param isVisibility visibility of the progress bar
```
Comment (ezaquarii) [inline src/main/java/com/owncloud/android/ui/activity/ToolbarActivity.java:166]: While here, we could actually correct the splling so it's not wonky: `isVisible`.

Comment (JorisBodin) [reply src/main/java/com/owncloud/android/ui/activity/ToolbarActivity.java:166]: For the next PR 😁

**Comment by AndyScherzinger** [comment]
Looks good to me 👍

Great work @JorisBrodin 🏆

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky what do you think how to much forward branching wise?
I'd be happy to see the UI changes going to master but the whole redesign is unlikely completed for 3.12. So we would have to create 3.12 rather soon and live with a small feature release 3.12 and see that the UI is done for 3.13.

What do you all think? @ezaquarii @JorisBrodin @Shagequi, I think we should make moving this forward as easy as possible where a separate branch would create a update hell regarding stating up to date with Master :/

**Comment by JorisBodin** [comment]
@AndyScherzinger 3.12 When is it scheduled?
Cause I think we're done for May 7

**Comment by ostasevych** [comment]
What about dark theme? Now it is not shown well, when some texts are also dark on the dark background?

**Comment by JorisBodin** [comment]
Hello @ostasevych
Do you have a exemple?

**Comment by AndyScherzinger** [comment]
>  3.12 When is it scheduled?

@JorisBodin see #5644

@tobiasKaminsky according to schedule today would be 3.12. RC1, so we would be on time for 3.13 with this (I guess?)

**Comment by JorisBodin** [comment]
> > 3.12 When is it scheduled?
> 
> @JorisBodin see #5644

I think the current modification of the appbar can be proposed in 3.12 :)

I modified the variable as suggested by @ezaquarii
And I also changed the ActionMode colors.

**Comment by JorisBodin** [comment]
Improve item selected background 
Before
Screenshot_1587990422
After
Screenshot_1587990333

**Comment by tobiasKaminsky** [comment]
@JorisBodin there are conflicts?
Hopefully I did not created them…

Regarding time frame:
Today I branched off 3.12, which will be released in ~2 weeks.
So we have enough time to get all our design changes in 3.13 :+1:

**Comment by JorisBodin** [comment]
@tobiasKaminsky The merge is now ok

I found that it was the wrong color of the back button in preview mode.
Screenshot_1588081058

I'll fix it later.

And https://github.com/nextcloud/android/pull/5951 doesn't work, so I took the liberty of removing it
Screenshot_1588081352

File: src/main/java/com/owncloud/android/ui/activity/FileDisplayActivity.java [hunk 2]
```diff
@@ -2227,7 +2227,7 @@ public void run() {
                                         null
                                 );
 
-                                setIndeterminate(true);
+                                showProgressBar(true);//okkk
```
Comment (tobiasKaminsky) [inline src/main/java/com/owncloud/android/ui/activity/FileDisplayActivity.java:2230]: Please remove this comment

File: src/main/java/com/owncloud/android/utils/ThemeUtils.java [hunk 3]
```diff
@@ -224,17 +218,18 @@ public static int fontColor(Context context) {
...
     /**
      * Tests if light color is set
-     * @return  true if primaryColor is lighter than MAX_LIGHTNESS
+     *
+     * @return true if primaryColor is lighter than MAX_LIGHTNESS
      */
-    public static boolean lightTheme(Context context) {
-        int primaryColor = primaryColor(context);
-        float[] hsl = colorToHSL(primaryColor);
+    public static boolean lightTheme(Context context, int color) {
```
Comment (tobiasKaminsky) [inline src/main/java/com/owncloud/android/utils/ThemeUtils.java:224]: This one now does not use context any longer.
Please remove it and change doc.

File: src/main/java/com/owncloud/android/utils/ThemeUtils.java [hunk 4]
```diff
@@ -244,6 +239,22 @@ public static boolean darkTheme(Context context) {
         return hsl[INDEX_LUMINATION] <= 0.55;
     }
 
+    public static int primaryAppbarColor(Context context) {
```
Comment (tobiasKaminsky) [inline src/main/java/com/owncloud/android/utils/ThemeUtils.java:242]: Is this background?
Then maybe appBarPrimaryBackgroundColor

File: src/main/java/com/owncloud/android/utils/ThemeUtils.java [hunk 5]
```diff
@@ -244,6 +239,22 @@ public static boolean darkTheme(Context context) {
...
+        return ContextCompat.getColor(context, R.color.appbar);
+    }
+
+    public static int fontAppbarColor(Context context) {
```
Comment (tobiasKaminsky) [inline src/main/java/com/owncloud/android/utils/ThemeUtils.java:246]: appBarPrimaryFontColor

File: src/main/java/com/owncloud/android/utils/ThemeUtils.java [hunk 6]
```diff
@@ -244,6 +239,22 @@ public static boolean darkTheme(Context context) {
...
+        return ContextCompat.getColor(context, R.color.fontAppbar);
+    }
+
+    public static int fontSecondaryAppbar(Context context) {
```
Comment (tobiasKaminsky) [inline src/main/java/com/owncloud/android/utils/ThemeUtils.java:250]: appBarSecondaryFontColor

Comment (tobiasKaminsky) [reply src/main/java/com/owncloud/android/utils/ThemeUtils.java:250]: (my idea is to first have the "where" and then "what".

File: src/main/java/com/owncloud/android/ui/activity/FileDisplayActivity.java (hunk 2 above)
Comment (JorisBodin) [reply src/main/java/com/owncloud/android/ui/activity/FileDisplayActivity.java:2230]: Oupsi 🙈

**Comment by tobiasKaminsky** [comment]
Some minor code formatting changes.
@jancborchardt have you tested the apk and are you fine with the design so far?

File: src/main/java/com/owncloud/android/utils/ThemeUtils.java (hunk 4 above)
Comment (JorisBodin) [reply src/main/java/com/owncloud/android/utils/ThemeUtils.java:242]: It's a point of view. But it's ok :)

**Comment by tobiasKaminsky** [comment]
Merged master again into it.
This should now work, and then can be merged.

**Comment by AndyScherzinger** [comment]
One minor issue that should be fixed (maybe in another PR?) is the intermediate bar which when shown/hidden makes the content jump, this didn't happen before.

Is this appbar shown anywhere where the loading needs to be displayed? Asking sind the search bar would rather use the circular spinner and maybe we should just switch to that one only, since we use it on other screens like activities or notifications.

**Pull request merged**